src/walletSdk/Horizon/Transaction/SponsoringBuilder.ts

```diff
@@ -0,0 +1,74 @@
+import StellarSdk, {
+  TransactionBuilder as StellarTransactionBuilder,
+  Transaction,
+} from "stellar-sdk";
+import { IssuedAssetId } from "../../Asset";
+
+import { CommonTransactionBuilder } from "./CommonTransactionBuilder";
+import { AccountKeypair } from "../Account";
+
+export class SponsoringBuilder extends CommonTransactionBuilder<SponsoringBuilder> {
+  private sponsorAccount: AccountKeypair;
+  private builder: StellarTransactionBuilder;
+
+  constructor(
+    sponsoredAddress: string,
+    sponsorAccount: AccountKeypair,
+    builder: StellarTransactionBuilder,
+  ) {
+    super(sponsoredAddress);
+    this.sponsorAccount = sponsorAccount;
+    this.builder = builder;
+
+    this.startSponsoring();
+  }
+
+  createAccount(
+    newAccount: AccountKeypair,
+    startingBalance: number = 0,
+  ): SponsoringBuilder {
+    this.operations.push(
+      StellarSdk.Operation.createAccount({
+        destination: newAccount.publicKey,
+        startingBalance: startingBalance.toString(),
+        source: this.sponsorAccount.publicKey,
+      }),
+    );
+    return this;
+  }
+
+  addAssetSupport(asset: IssuedAssetId, trustLimit?: string) {
+    super.addAssetSupport(asset, trustLimit);
+    return this;
+  }
+
+  removeAssetSupport(asset: IssuedAssetId) {
```

Comment:
could we add return type here?

Reply:
https://github.com/stellar/typescript-wallet-sdk/pull/45/commits/df09183d3e27553519ffc953efc023d7068c407d